From 939e89458b6ede421153daced7cffa56f497393c Mon Sep 17 00:00:00 2001
From: Minecrell <minecrell@minecrell.net>
Date: Mon, 9 Dec 2019 22:54:36 +0100
Subject: [PATCH 063/210] MSM8916: ARM: configs: Add additional config options
 for MSM8916 on ARM32 (v3)

The kernel configuration is largely shared with ARM64, so these
are only the additional options needed for ARM32. It needs to be
concatenated to the ARM64 msm8916_defconfig (see comment).

v2: Disable CONFIG_ARM_SMMU on ARM32 since in practice it's only useful
with 64-bit custom TZ.
v3: CONFIG_ARM_PSCI=y for DB410c
---
 arch/arm/configs/.gitignore             |  2 ++
 arch/arm/configs/msm8916_defconfig.part | 32 +++++++++++++++++++++++++
 2 files changed, 34 insertions(+)
 create mode 100644 arch/arm/configs/.gitignore
 create mode 100644 arch/arm/configs/msm8916_defconfig.part

diff --git a/arch/arm/configs/.gitignore b/arch/arm/configs/.gitignore
new file mode 100644
index 00000000000..1cd39ba800c
--- /dev/null
+++ b/arch/arm/configs/.gitignore
@@ -0,0 +1,2 @@
+# This is dynamically built from the arm64 one (see msm8916_defconfig.part)
+msm8916_defconfig
diff --git a/arch/arm/configs/msm8916_defconfig.part b/arch/arm/configs/msm8916_defconfig.part
new file mode 100644
index 00000000000..8da5974424c
--- /dev/null
+++ b/arch/arm/configs/msm8916_defconfig.part
@@ -0,0 +1,32 @@
+# Build the full configuration by running
+# cat arch/arm64/configs/msm8916_defconfig arch/arm/configs/msm8916_defconfig.part > arch/arm/configs/msm8916_defconfig
+
+CONFIG_PERF_EVENTS=y
+CONFIG_ARCH_MSM8916=y
+CONFIG_SMP=y
+CONFIG_ARM_PSCI=y
+CONFIG_HZ_250=y
+CONFIG_HIGHMEM=y
+# CONFIG_ATAGS is not set
+CONFIG_VFP=y
+CONFIG_NEON=y
+CONFIG_KERNEL_MODE_NEON=y
+CONFIG_ARM_CRYPTO=y
+CONFIG_CRYPTO_SHA1_ARM_NEON=y
+CONFIG_CRYPTO_SHA1_ARM_CE=y
+CONFIG_CRYPTO_SHA2_ARM_CE=y
+CONFIG_CRYPTO_SHA512_ARM=m
+CONFIG_CRYPTO_AES_ARM=y
+CONFIG_CRYPTO_AES_ARM_BS=y
+CONFIG_CRYPTO_AES_ARM_CE=y
+CONFIG_CRYPTO_GHASH_ARM_CE=y
+CONFIG_CRYPTO_CRC32_ARM_CE=y
+CONFIG_GPIOLIB=y
+CONFIG_POWER_RESET=y
+CONFIG_POWER_SUPPLY=y
+CONFIG_CRYPTO=y
+CONFIG_CRYPTO_SHA3=m
+CONFIG_CRYPTO_SM3=m
+CONFIG_CRYPTO_SM4=m
+CONFIG_STRICT_DEVMEM=y
+# CONFIG_ARM_SMMU is not set
-- 
2.34.1

